### PR TITLE
8.x 2.x  2757701  enforce unique attribute in product variations

### PR DESCRIPTION
**Pull request by ndf**
Initial setup with a failing test and boilerplate for the attribute Constraint class

### CODE REVIEW

File: modules/product/src/Plugin/Validation/Constraint/ProductVariationAttributeConstraintValidator.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * Constructs a new ProductVariationTypeForm object.
```
Comment (steveoliver) [inline modules/product/src/Plugin/Validation/Constraint/ProductVariationAttributeConstraintValidator.php:24]: wrong class name

File: modules/product/src/Plugin/Validation/Constraint/ProductVariationAttributeConstraint.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+/**
+ * Ensures product variation attribute uniqueness.
+ *
+ * @Constraint(
+ *   id = "ProductVariationAttribute",
+ *   label = @Translation("The attribute of the product variation.", context = "Validation"),
+ *   type = "entity:commerce_product_variation"
+ * )
+ */
+class ProductVariationAttributeConstraint extends CompositeConstraintBase {
```
Comment (steveoliver) [inline modules/product/src/Plugin/Validation/Constraint/ProductVariationAttributeConstraint.php:16]: maybe call it ProductVariationUniqueAttributeConstraint for clarity?

**Review by steveoliver — CHANGES_REQUESTED** [review]
Would be great to have this in.  Let's the get [coding standards issues](https://travis-ci.org/drupalcommerce/commerce/jobs/142247069#L970) fixed + consider my comments for changes.

**Comment by mglaman** [comment]
@steveoliver this is pretty stale (pre-beta, even) from Drupal Dev Days. Maintainers can push to it, but it might be best to export current work to a patch on Drupal.org, then carry it on in a new PR (since only maintainers can push.)

**Comment by bojanz** [comment]
Closing, since the PR is severely outdated

**Pull request closed without merging**